### PR TITLE
[BACKLOG-41091] Decode PVFS File Paths: Scheduler Perspective + Browse Perspective missed case

### PR DESCRIPTION
**Pull request by singletonc**
- Move decode logic into URLEncoder.js for general use in project

PR Set:
1. https://github.com/pentaho/pentaho-platform-plugin-common-ui/pull/1777
2. https://github.com/pentaho/pentaho-commons-gwt-modules/pull/1041
3. https://github.com/pentaho/pentaho-platform/pull/5647
4. https://github.com/pentaho/pentaho-scheduler-plugin/pull/183

### CODE REVIEW

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 9m 28s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox
```
---
:ok_hand: All tests passed!
 
Tests run: **5241**, Failures: **0**, Skipped: **5** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/common-ui/job/PR-1777/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by rmansoor — APPROVED** [review]

File: impl/client/src/main/javascript/web/util/URLEncoder.js [hunk 1]
```diff
@@ -123,6 +123,19 @@ define("common-ui/util/URLEncoder", [], function() {
...
+    try {
+      return decodeURI(attributeValue).replace("%23", "#");
+    } catch (error) {
+      // if there is an error, simply return the value. This should only impact the UI visually, not functionally.
+      // we can show an error if we'd like, but have opted not to at this time.
+      // let errorMessage = "Error decoding file attributeValue: " + attributeValue + "\n" + error;
+      // window.parent.mantle_showGenericError(errorMessage);
+      return attributeValue;
+    }
+  }
```
Comment (dcleao) [inline impl/client/src/main/javascript/web/util/URLEncoder.js:138]: Not needed in solution proposed in: https://github.com/pentaho/pentaho-platform/pull/5647#pullrequestreview-2151052689.

**Review by dcleao — CHANGES_REQUESTED** [review]

**Comment by singletonc** [comment]
Closing in favor of new PR Set:
[pentaho/pentaho-commons-gwt-modules#1042](https://github.com/pentaho/pentaho-commons-gwt-modules/pull/1042)
[pentaho/pentaho-platform#5652](https://github.com/pentaho/pentaho-platform/pull/5652)
[pentaho/pentaho-scheduler-plugin#187](https://github.com/pentaho/pentaho-scheduler-plugin/pull/187)
[pentaho/pentaho-scheduler-plugin-ee#256](https://github.com/pentaho/pentaho-scheduler-plugin-ee/pull/256)

**Pull request closed without merging**